phase08/StudentGrades/StudentGrades/Student.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace project;
+
+public record Student
+{
+    public int StudentNumber { get; init; }
+    public string FirstName { get; init; }
+    public string LastName { get; init; }
+    public double AverageGrade { get; set; }
```

Comment:
If you want to keep the average grade for each student it must be updated after a new lesson is inserted. To fix this issue you can call `CalculateAverageGrade` for all students in `DBController` class in `AddRecordsToDB` or just call it for students with a new grade record inserted. this is done in `O(n)`, `n` being the number of grades. but the best way is to keep track of number of grades for each student so when a new grade is added average could be calculated in `O(1)`.